### PR TITLE
Move scripts used in CI to the new location

### PR DESCRIPTION
**Pull request by sergejparity**
This PR moves CI related scripts and helper files to the new location **./scripts/ci**.
We are performing this to unify structure of all paritytech repositories. Target is to improve management of the CI related stuff.
Related issue: https://github.com/paritytech/ci_cd/issues/313
Also this is reflected in article: Policies and Processes: [Where do I start my new project?](https://github.com/paritytech/ci_cd/wiki/Policies-and-Processes:-Where-do-I-start-my-new-project%3F)

### CODE REVIEW

**Comment by dviejokfs** [comment]
@sergejparity I can't access this repository: https://github.com/paritytech/ci_cd

Is this repository private?

**Review by radupopa2010 — APPROVED** [review]

**Comment by sergejparity** [comment]
> @sergejparity I can't access this repository: https://github.com/paritytech/ci_cd
> 
> Is this repository private?

@dviejokfs yes it is CI team internal project.

**Review by alvicsam — APPROVED** [review]
Nice work, LGTM

File: docs/node-template-release.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ the existence of your current git commit ID in the remote repository.
 	Assume you are in root directory of Substrate. Run:
 
 	```bash
-	cd .maintain/
+	cd .scripts/ci/
```
Comment (joao-paulo-parity) [inline docs/node-template-release.md:10]:
```suggestion
	cd scripts/ci/
```

**Review by joao-paulo-parity — CHANGES_REQUESTED** [review]

File: scripts/ci/node-template-release/Cargo.lock [hunk 2]
```diff
@@ -0,0 +1,578 @@
+# This file is automatically @generated by Cargo.
```
Comment (joao-paulo-parity) [inline scripts/ci/node-template-release/Cargo.lock:1]: https://github.com/paritytech/substrate/tree/master/.maintain/node-template-release does not include a Cargo.lock, perhaps it is on purpose?

Comment (bkchr) [reply scripts/ci/node-template-release/Cargo.lock:1]: Yeah this file should be removed.

Comment (sergejparity) [reply scripts/ci/node-template-release/Cargo.lock]: Removed.

**Review by joao-paulo-parity — APPROVED** [review]

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -185,7 +185,7 @@ default:
           tee ./artifacts/substrate/VERSION;
     fi
   - sha256sum ./artifacts/substrate/substrate | tee ./artifacts/substrate/substrate.sha256
-  - cp -r .maintain/docker/substrate.Dockerfile ./artifacts/substrate/
+  - cp -r scripts/ci/docker/substrate.Dockerfile ./artifacts/substrate/
```
Comment (TriplEight) [inline .gitlab-ci.yml:188]:
```suggestion
  - cp -r ./scripts/ci/docker/substrate.Dockerfile ./artifacts/substrate/
```
let's be consistent with paths

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -288,11 +288,11 @@ test-prometheus-alerting-rules:
     - if: $CI_COMMIT_BRANCH
       changes:
         - .gitlab-ci.yml
-        - .maintain/monitoring/**/*
+        - scripts/ci/monitoring/**/*
```
Comment (TriplEight) [inline .gitlab-ci.yml:291]:
```suggestion
        - ./scripts/ci/monitoring/**/*
```

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -288,11 +288,11 @@ test-prometheus-alerting-rules:
...
   script:
-    - promtool check rules .maintain/monitoring/alerting-rules/alerting-rules.yaml
-    - cat .maintain/monitoring/alerting-rules/alerting-rules.yaml |
-        promtool test rules .maintain/monitoring/alerting-rules/alerting-rule-tests.yaml
+    - promtool check rules scripts/ci/monitoring/alerting-rules/alerting-rules.yaml
```
Comment (TriplEight) [inline .gitlab-ci.yml:293]:
```suggestion
    - promtool check rules ./scripts/ci/monitoring/alerting-rules/alerting-rules.yaml
```

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -288,11 +288,11 @@ test-prometheus-alerting-rules:
...
+    - cat scripts/ci/monitoring/alerting-rules/alerting-rules.yaml |
```
Comment (TriplEight) [inline .gitlab-ci.yml:294]:
```suggestion
    - cat ./scripts/ci/monitoring/alerting-rules/alerting-rules.yaml |
```

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -288,11 +288,11 @@ test-prometheus-alerting-rules:
...
+        promtool test rules scripts/ci/monitoring/alerting-rules/alerting-rule-tests.yaml
```
Comment (TriplEight) [inline .gitlab-ci.yml:295]:
```suggestion
        promtool test rules ./scripts/ci/monitoring/alerting-rules/alerting-rule-tests.yaml
```

File: .gitlab-ci.yml [hunk 8]
```diff
@@ -301,10 +301,10 @@ cargo-deny:
   <<:                              *docker-env
   <<:                              *nightly-pipeline
   script:
-    - cargo deny check --hide-inclusion-graph -c .maintain/deny.toml
+    - cargo deny check --hide-inclusion-graph -c scripts/ci/deny.toml
```
Comment (TriplEight) [inline .gitlab-ci.yml:304]:
```suggestion
    - cargo deny check --hide-inclusion-graph -c ./scripts/ci/deny.toml
```

File: .gitlab-ci.yml [hunk 9]
```diff
@@ -301,10 +301,10 @@ cargo-deny:
...
   after_script:
     - echo "___The complete log is in the artifacts___"
-    - cargo deny check -c .maintain/deny.toml 2> deny.log
+    - cargo deny check -c scripts/ci/deny.toml 2> deny.log
```
Comment (TriplEight) [inline .gitlab-ci.yml:307]:
```suggestion
    - cargo deny check -c ./scripts/ci/deny.toml 2> deny.log
```

File: .gitlab-ci.yml [hunk 10]
```diff
@@ -590,7 +590,7 @@ build-linux-subkey:                &build-subkey
         sed -n -E 's/^subkey ([0-9.]+.*)/\1/p' |
           tee ./artifacts/subkey/VERSION;
     - sha256sum ./artifacts/subkey/subkey | tee ./artifacts/subkey/subkey.sha256
-    - cp -r .maintain/docker/subkey.Dockerfile ./artifacts/subkey/
+    - cp -r scripts/ci/docker/subkey.Dockerfile ./artifacts/subkey/
```
Comment (TriplEight) [inline .gitlab-ci.yml:593]:
```suggestion
    - cp -r ./scripts/ci/docker/subkey.Dockerfile ./artifacts/subkey/
```

File: .gitlab-ci.yml [hunk 11]
```diff
@@ -807,4 +807,4 @@ deploy-prometheus-alerting-rules:
     - if: $CI_COMMIT_REF_NAME == "master"
       changes:
         - .gitlab-ci.yml
-        - .maintain/monitoring/**/*
+        - scripts/ci/monitoring/**/*
```
Comment (TriplEight) [inline .gitlab-ci.yml:810]:
```suggestion
        - ./scripts/ci/monitoring/**/*
```

**Review by TriplEight — APPROVED** [review]
I think it'll be easier to check once again what's left after this is merged.

**Comment by ggwpez** [comment]
The label check keeps failing:  
```pre
bash: /home/runner/work/substrate/substrate/scripts/ci/github/check_labels.sh: No such file or directory
```
https://github.com/paritytech/substrate/runs/5536013763?check_suite_focus=true

**Comment by sergejparity** [comment]
> The label check keeps failing:
> 
> ```
> bash: /home/runner/work/substrate/substrate/scripts/ci/github/check_labels.sh: No such file or directory
> ```
> 
> https://github.com/paritytech/substrate/runs/5536013763?check_suite_focus=true

You. Need to update your branch with latest `master` as location of CI helper scripts has changed.

**Pull request merged**